views/metrilo-import-view.php

```diff
@@ -1,116 +1,147 @@
+                        break;
+                    case 'deletedProducts':
+                        import_type(newChunkId, dataOptions, 'deletedProducts', 'products');
+                        break;
+                    case 'products':
+                        import_type(newChunkId, dataOptions, 'products', 'orders');
+                        break;
+                    case 'orders':
+                        import_type(newChunkId, dataOptions, 'orders', null);
+                        break;
+                    default:
+                        return false;
                 }
+            });
+        }
 
-                function import_type(newChunkId, data, current, next) {
-                    if (data[`${current}Chunks`] > 0) {
-                        data.percentage = (100 / data[`${current}Chunks`]);
-                    }
-                    var hasMoreChunks = newChunkId < data[`${current}Chunks`];
+        function ajax_post_with_retry(url, data, callback) {
+            if (dataOptions.retryAttempts !== 0) {
+                jQuery.post(url, data, function (response) {
+                    callback();
+                }).fail(function () {
+                    dataOptions.retryAttempts -= 1;
+                    setTimeout(function () {
+                        ajax_post_with_retry(url, data, callback);
+                    }, 5000);
+                })
+            } else {
+                dataOptions.retryAttempts = 3;
+                sync_chunk(data.chunkId + 1, data.importType, true);
+            }
+        }
 
-                    if(hasMoreChunks) {
-                        this.sync_chunk(newChunkId, data.importType, false);
-                    } else {
-                        if(current == 'orders') {
-                            this.update_importing_message("<span style='color: green;'>" + 'Done! Please expect up to 30 minutes for your historical data to appear in Metrilo.' + "</span>", true);
-                            jQuery('#metrilo_import_step').hide();
-                        } else {
-                            jQuery('#metrilo_import_step').text((`Importing ${next}`));
-                            dataOptions.importType = next;
-                            this.sync_chunk(0, dataOptions.importType, false);
-                        }
-                    }
-                }
+        function import_type(newChunkId, data, current, next) {
+            if (data[`${current}Chunks`] > 0) {
+                data.percentage = (100 / data[`${current}Chunks`]);
+            }
+            var hasMoreChunks = newChunkId < data[`${current}Chunks`];
 
-                function update_importing_message(message, show_loader) {
-                    if (show_loader) {
-                        jQuery('#metrilo_import_status').html(message);
-                    }
+            if (hasMoreChunks) {
+                sync_chunk(newChunkId, data.importType, false);
+            } else {
```

Comment:
you can use a guard clause(s) here as well, It could go like:

```
if (hasMoreChunks) { //do stuff, return }
if(current == 'orders') { // do stuff, return }

// importing next comes here
```

Reply:
Refactored.